debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/AbstractLogMinerEventProcessor.java

```diff
@@ -1419,24 +1420,35 @@ public void abandonTransactions(Duration retention) throws InterruptedException
                 if (!smallestScn.isNull() && thresholdScn.compareTo(smallestScn) >= 0) {
                     boolean first = true;
                     Iterator<Map.Entry<String, T>> iterator = getTransactionCache().entrySet().iterator();
-                    while (iterator.hasNext()) {
-                        Map.Entry<String, T> entry = iterator.next();
-                        if (entry.getValue().getStartScn().compareTo(thresholdScn) <= 0) {
-                            if (first) {
-                                LOGGER.warn("All transactions with SCN <= {} will be abandoned.", thresholdScn);
-                                first = false;
+                    try {
+                        while (iterator.hasNext()) {
+                            Map.Entry<String, T> entry = iterator.next();
+                            if (entry.getValue().getStartScn().compareTo(thresholdScn) <= 0) {
+                                if (first) {
+                                    LOGGER.warn("All transactions with SCN <= {} will be abandoned.", thresholdScn);
+                                    LOGGER.debug("List of transactions in the cache before transactions being abandoned: [{}]",
+                                            getTransactionCache().keySet().stream().collect(Collectors.joining(",")));
```

Comment:
Guard this with `if(LOGGER.isDebugEnabled())` to avoid the expensive keySet collection if debugging isn't enabled.